integration_test/tests/generating.rs

```diff
@@ -8,6 +8,39 @@ use integration_test::{Node, NodeExt as _, Wallet};
 use node::vtype::*;             // All the version specific types.
 use node::mtype;
 
+#[cfg(not(feature = "v20_and_below"))]
+#[test]
+fn generating__generate_block__modelled() {
+    let node = Node::with_wallet(Wallet::Default, &[]);
+    node.fund_wallet();
+    let mining_addr = node.client.new_address().expect("failed to get new address");
+    let dest_addr = node.client.new_address().expect("failed to get new address");
+    let amount = bitcoin::Amount::from_sat(1_000_000);
+    let txid = node
+        .client
+        .send_to_address_rbf(&dest_addr, amount)
+        .expect("sendtoaddressrbf")
+        .txid()
+        .expect("txid");
+    let transactions = vec![txid.to_string()];
+
+    #[cfg(feature = "v24_and_below")]
+    {
+        // No `submit` argument
+        let json: GenerateBlock = node.client.generate_block(&mining_addr.to_string(), &transactions).expect("generateblock");
+        let model: Result<mtype::GenerateBlock, GenerateBlockError> = json.into_model();
+        model.unwrap();
+    }
+
+    #[cfg(not(feature = "v24_and_below"))]
+    {
+        // Check with `submit = false` so that `hex` is returned. v25 and later only.
+        let json: GenerateBlock = node.client.generate_block(&mining_addr.to_string(), &transactions, false).expect("generateblock");
+        let model: Result<mtype::GenerateBlock, GenerateBlockError> = json.into_model();
+        model.unwrap();
+    }
```

Comment:
Can do as a follow up. I rekon the feature guards could just guard the `generate_block` call. The other two lines are identical.